com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/ResourceManager.java

```diff
@@ -90,20 +91,44 @@ private void registerExtensions(Set<EPackageConfiguration> configurations) {
 		return loadResource(URI.createURI(resourceURI), rs);
 	}
 
-	public <T extends EObject> Optional<T> loadModel(URI resourceURI, ResourceSet rs, Class<T> clazz) {
+	/**
+	 * Attempts to find a file resource in a casa insensitive way.
+	 */
+	private static String findResource(String filename)	{
+		File file = new File(filename);
+		File directory = file.getParentFile();
+		String[] files = directory.list();
+
+		if (files != null) {
+			for (String f : files) {
+				if (f.equalsIgnoreCase(file.getName())) {
+					return directory.getAbsolutePath() + "/" + f;
+				}
+			}
+		}
+
+		return filename;
+	}
+
+	public <T extends EObject> Optional<LoadModelResult<T>> loadModel(URI resourceURI, ResourceSet rs, Class<T> clazz) {
```

Comment:
This should return a `Resource` from a `ResourceSet`, which maybe now should be managed by the `ResourceManager`?  I'm not sure how the responsibilities of resource management are divided between the `ResourceManager` and the `ModelRepository`.